### PR TITLE
Refactor: Distinguish between two concepts previously called "Dummy"

### PR DESCRIPTION
**Pull request by heinrich5991**
On the one hand variables called "Dummy" would tell us whether the
current action refers to the currently inactive tee ("dummy"). On the
other hand, these variables could tell us whether the current action
refers to the main connection to the server, or the secondary one. The
latter use case is now renamed to "Client", with the choices
`CLIENT_MAIN`, `CLIENT_DUMMY` (and `CLIENT_CONTACT`).

Perhaps better names could be found, especially since `Client` also
refers to the engine client class in the game code.

Also fix a few bugs in dummy handling.

## Checklist

- [x] Tested the change ingame
- [ ] Provided screenshots if it is a visual change
- [ ] Tested in combination with possibly related configuration options
- [ ] Written a unit test if it works standalone, system.c especially
- [ ] Considered possible null pointers and out of bounds array indexing
- [ ] Changed no physics that affect existing maps
- [ ] Tested the change with [ASan+UBSan or valgrind's memcheck](https://github.com/ddnet/ddnet/#using-addresssanitizer--undefinedbehavioursanitizer-or-valgrinds-memcheck) (optional)

### CODE REVIEW

**Comment by heinrich5991** [comment]
> Perhaps better names could be found, especially since `Client` also refers to the engine client class in the game code.

Ideas?

**Comment by def-** [comment]
Connection/CONN? Since it's not just main & dummy, but also Contact.

**Comment by heinrich5991** [comment]
> Connection/CONN? Since it's not just main & dummy, but also Contact.

Very nice idea, thanks!

Done.

File: src/engine/client/client.cpp [hunk 1]
```diff
@@ -1822,45 +1812,39 @@ void CClient::ProcessServerPacket(CNetChunk *pPacket)
 				m_pConsole->Print(IConsole::OUTPUT_LEVEL_STANDARD, "client", aBuf);
 			}
 		}
-		else if((pPacket->m_Flags & NET_CHUNKFLAG_VITAL) != 0 && Msg == NETMSG_RCON_CMD_ADD)
+		else if(!Dummy && (pPacket->m_Flags & NET_CHUNKFLAG_VITAL) != 0 && Msg == NETMSG_RCON_CMD_ADD)
```
Comment (fokkonaut) [inline src/engine/client/client.cpp:1815]: This can probably lead to double entries while swapping again

**Pull request merged**